### ISSUE TITLE
save method in method in UnPackaged-Globals got 'expected Character Collection error'

### ISSUE THREAD

**Issue by ericwinger** [dalehenrich/Rowan#849]
Not sure if this is a reproduction case, but here is what I did right before getting the error. 

Using Jadeite 3.2.9, RowanClientServices 3.0.1

- I created a class, `TestBP` in the UnPackaged-Globals package. 
- Created this method on the instance side
```
methodToStopIn

	| abc theSize |
	abc := OrderedCollection new. 
	abc add: 'foo'; add: #foo. 
	theSize := abc size. 
	theSize = 2 ifTrue:[abc add: 400]. 
	^abc
```
- Put a breakpoint in it, did some debugger testing. 
- Changed the method source to what you see below. 
- Saved the method. 
- Got walkback `a ArgumentTypeError occurred (error 2094), reason:rtErrBadArgKind, expected a CharacterCollection`

```
methodToStopIn

	| abc theSize |
	abc := OrderedCollection new. 
	abc add: 'foo'; add: #foo. 
	theSize := abc size. 
	abc do:[:thing |
	|blah |
	blah := thing]. 
	theSize = 2 ifTrue:[abc add: 400]. 
	^abc
```

[20220819_expect_CharacterCollection_rowan_walkback.txt](https://github.com/GemTalk/Rowan/files/9384785/20220819_expect_CharacterCollection_rowan_walkback.txt)

The method source argument was a String.

**Comment by ericwinger** [comment]
Here's what my browser looked like in case my issue report isn't clear enough. 
![image](https://user-images.githubusercontent.com/36212516/185691589-e539c7a3-437d-4ac5-be20-e40301b70472.png)

**Comment by ericwinger** [comment]
Looks like somehow, my `RwGsPatchSet_V2>>loadSymbolList` has two symbol dictionaries without names in them. That seems to be the immediate problem in `RwGsPatchSet_V2>>_createNewSymbolList`. Not sure why the method's temp vars have their own symbol dictionary though.

```
1: aSymbolDictionary( #'theSize'->nil, #'abc'->anOrderedCollection( ))
2: aSymbolDictionary( #'theSize'->nil, #'abc'->anOrderedCollection( 'foo', #'foo', 400))
3: aSymbolDictionary( name: #'UserGlobals' )
4: aSymbolDictionary( name: #'Globals' )
5: aSymbolDictionary( name: #'Published' )
6: aSymbolDictionary( name: #'RowanKernel' )
7: aSymbolDictionary( name: #'RowanLoader' )
8: aSymbolDictionary( name: #'RowanTools' )
9: aSymbolDictionary( name: #'RowanClientServices' )
```

**Comment by dalehenrich** [comment]
I don't think that we need to worry about interacting with UNPACKGED guys they are going away and any problems will go away with them :)

**Comment by rjsargent** [comment]
I dealt with this same issue just a moment ago.
It turns out that if you evaluate an expression from the debugger window, the debugger adds a new symbol dictionary to the front of the symbol list with the (I think) variables in the frame. The problem is that the symbol dictionary is unnamed. RwGsPatchSet_V2 >> _createNewSymbolList thinks only named symbol dictionaries will be found. This is not the first time I have seen problems that assume every symbol dictionary is named.

I think having the debugger name the symbol dictionary might be a good idea anyway. e.g. a combination of the word debugger and the process' id, so that multiple debuggers won't be fighting over what they think is theirs.

**Comment by ericwinger** [comment]
Ahh, so Jadeite might be contributing. As you suggested, I added a name to the symbol dictionary added by the debugger during an ad-hoc execution. The name is 'DebuggerExecution<process id>'. 

I'll close this issue.

**Comment by dalehenrich** [comment]
reopening and assigning to me ... I am surprised to see that symbol dictionaries are creating symbol dictionaries on the fly and am wondering about whether this makes sense or not, especially when `RwGsPatchSet_V2 >> _createNewSymbolList` is involved in ad-hoc execution ... not a red flag, but the yellow light is blinking :smile:
